src/components/tab-group/tab-group.component.ts

```diff
@@ -365,6 +371,21 @@ export default class SlTabGroup extends ShoelaceElement {
     return nextTab;
   }
 
+  private updateScrollButtons() {
+    if (this.hasScrollControls && this.autoHideScrollButtons) {
+      this.hideScrollStartButton = !this.doesHaveTabsPassedScrollStart();
+      this.hideScrollEndButton = !this.doesHaveTabsPassScrollEnd();
+    }
+  }
+
+  private doesHaveTabsPassedScrollStart() {
```

Comment:
- [ ] Consider `canScrollLeft` and `canScrollRight`
Also the properties can be renamed
- [ ] Defining a method is often a preferred means of creating abstraction vs variable names. Here, though
where we are simply setting property values anyway, I'm not so sure that the methods are pulling their weight.
- [ ] The `Math.abs` is here to support RTL, but that isn't made so clear. Consider moving the functions `doesHaveTabsPassedScrollStart` and `doesHaveTabsPassedScrollEnd` inline, and extracting the scrollLeft/abs logic to a seperate method, with clear name, and jsdoc comment.
Also, this doesn't handle the possibility of overscrolling. IDK if that's possible in this context, but consider being more explicit with `this.localize.dir()` (See `handleScrollTo(Start|End)`). That becomes more reasonable when the logic of getting scrollLeft (scrollStart?) is moved to its own method.

Reply:
Ok I updated it. Hopefully i got everything.,, let me know if i'm missing anything